File: server/Server/Model/Device.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System.Collections.Generic;

namespace Server.Models
{
    [BsonIgnoreExtraElements]
    public class Device
    {
        public string Id { get; set; }
        public string DeviceId { get; set; }
        public string DeviceCode { get; set; }
        public float FreezingPoint { get; set; }
        public float BoilingPoint { get; set; }
        public List<TemperatureReading> TemperatureReadings { get; set; }
    }
}

File: app/src/app/_models/device.ts
import { Message } from "./message";
import { TemperatureReading } from "./temperatureReading";

export interface Device {
    deviceId: string,
    deviceCode: string,

    freezingPoint: number,
    boilingPoint: number,

    temperatureReadings: TemperatureReading[],

    alertMessages?: Message[],
    isShowWarnings?: boolean
  }
  